internal/connection.go

```diff
@@ -178,21 +176,21 @@ func (c *Connection) close(err error) {
 	}
 	close(c.closed)
 	c.closedTime.Store(time.Now())
-	c.connectionManager.connectionClosed(c, err)
+	c.connectionManager.onConnectionClose(c, err)
 }
 
 func (c *Connection) String() string {
 	return fmt.Sprintf("ClientConnection{"+
 		"isAlive=%t"+
 		", connectionID=%d"+
-		", endpoint=%s:%d"+
+		", endpoint=%s"+
 		", lastReadTime=%s"+
 		", lastWriteTime=%s"+
 		", closedTime=%s"+
 		", lastHeartbeatRequested=%s"+
 		", lastHeartbeatReceived=%s"+
 		", connected server version=%s", c.isAlive(), c.connectionID,
-		c.endpoint.Load().(*protocol.Address).Host(), c.endpoint.Load().(*protocol.Address).Port(),
+		c.endpoint.Load().(core.Address).String(),
```

Comment:
I dont think `.String()` is necessary here

Reply:
Interesting. Why it is used for every other field here ? I am removing all of them then.

Reply:
Yeah `String()` method is already called when trying to print a struct etc.